airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/source.py

```diff
@@ -80,6 +80,7 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
         # It doesnt support pagination so there is no sense of reading single
         # record, it would fetch all the data anyway.
         Profiles(config, authenticator=self._make_authenticator(config)).get_all_profiles()
```

Comment:
I'd suggest to move the function `get_all_profiles` to `_choose_profiles` and throw an error if doesn't find any profile whatever has data in `profiles` or `marketplaceid` in the config.

Reply:
It looks like during unit tests, `Profiles(config, authenticator=self._make_authenticator(config)).get_all_profiles()` returns an empty list. Adding logic to check filtering and raising an error if anything is empty breaks all unit tests. I will take a further look to see if this can be resolved.

Reply:
@marcosmarxm I just committed the logic for this, but I know the tests are going to fail until we can mock the unit tests.